### DISCUSSION TITLE
Error when deploying to Heroku

### DISCUSSION THREAD

**Discussion by benjamin-gambling** [prisma/prisma#3754] [Q&A] [answered]
Hello!

i created a hackernews clone using create-react-app and apollo on the front end and prisma, graphql-yoga and node on the backend.  i have tried to deploy with Heroku but am coming across this error that i just cant seem to get my head around and fix. i don't have a .yml file that it asks for with prisma generate, im wondering if thats an older version as the work i have done with prisma i have never been required to do that. 

i followed the howtographql tutorial but altered it a bit by adding a database that would update after deployment and i used tyepscript. The server on the front end tutorial is different to the server i created when doing the back end tutorial, which does have a. yml file. 

this is the error when i log $heroku logs --tail

```
2020-09-24T19:28:56.421126+00:00 app[web.1]: /app/node_modules/.prisma/client/index.js:3
2020-09-24T19:28:56.421145+00:00 app[web.1]: throw new Error(
2020-09-24T19:28:56.421146+00:00 app[web.1]: ^
2020-09-24T19:28:56.421146+00:00 app[web.1]: 
2020-09-24T19:28:56.421147+00:00 app[web.1]: Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
2020-09-24T19:28:56.421148+00:00 app[web.1]: In case this error is unexpected for you, please report it in https://github.com/prisma/prisma-client-js/issues/390.
2020-09-24T19:28:56.421148+00:00 app[web.1]: at new PrismaClient (/app/node_modules/.prisma/client/index.js:3:11)
```

Any help would be much appreciated! 

Source code: https://github.com/benjamin-gambling/hackernews

**Comment by ryands17** [comment, marked as answer]
Hey @benjamin-gambling 👋 

Have you followed [this](https://www.prisma.io/docs/guides/deployment/deploying-to-heroku) tutorial from our docs?

Also, your error seems that you either have not installed `@prisma/cli` and `@prisma/client` locally or you have an older version of Prisma downloaded someplace. Could you check that and let me know.

**Comment by richardwardza** [comment]
I've hooked Heroku up to my `prisma/graphql` api repo (so not create-react-app)

I've had to add the `heroku/nodejs` buildpack (on the heroku `settings` page).
The I have the following in `scripts` in my `package.json`:
```
        "postinstall": "yarn prisma generate && yarn prisma migrate up --experimental ",
```

When I do a `git push` Heroku does the deploy automatically (as expected)

**Comment by benjamin-gambling** [comment]
Thanks @ryands17 && @richardwardza !

I looked through it again after you suggestions and realized it was trying to install it when deploying in the root folder but my prisma folder was in a server folder, i did have a package.json in there for it to catch it but seemed it wanted it in the root folder so just moved it over and it worked! Had a couple of minor errors after that but got there in the end. Much appreciated for your time, if you guys hadn't replied i may have just given up :)  

Have a great day!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
